src/phatjs.rs

```diff
@@ -0,0 +1,142 @@
+#![cfg_attr(not(feature = "native"), no_main)]
+
+extern crate alloc;
+
+use js::ToJsValue;
+
+use anyhow::{anyhow, bail, Context, Result};
+use sidevm_quickjs::{runtime, Service};
+
+use pink_types::js::{JsCode, JsValue};
+
+struct Args {
+    codes: Vec<JsCode>,
+    js_args: Vec<String>,
+}
+
+fn parse_args() -> Result<Args> {
+    let mut codes = vec![];
+    let mut iter = std::env::args();
+    iter.next();
+    while let Some(arg) = iter.next() {
+        if arg.starts_with("-") {
+            if arg == "--" {
+                break;
+            }
+            match arg.as_str() {
+                "-c" => {
+                    let code = iter.next().ok_or(anyhow!("Missing code after -c"))?;
+                    codes.push(JsCode::Source(code));
+                }
+                "-b" => {
+                    let code = iter.next().ok_or(anyhow!("Missing code after -b"))?;
+                    let bytecode = hex::decode(code).context("Failed to decode bytecode")?;
+                    codes.push(JsCode::Bytecode(bytecode));
+                }
+                _ => {
+                    print_usage();
+                    bail!("Unknown option: {}", arg);
+                }
+            }
+        } else {
+            // File name
+            let code = std::fs::read_to_string(arg).context("Failed to read script file")?;
+            codes.push(JsCode::Source(code));
+        }
+    }
+    if codes.is_empty() {
+        print_usage();
+        bail!("No script file provided");
+    }
+    let js_args = iter.collect();
+    Ok(Args { codes, js_args })
+}
+
+fn print_usage() {
+    println!("Usage: phatjs [options] [script..] [-- [args]]");
+    println!("");
+    println!("Options:");
+    println!("  -c <code>        Execute code");
+    println!("  -b <hexed code>  Execute bytecode");
+    println!("  --               Stop processing options");
+}
+
+async fn run() -> Result<JsValue> {
+    let args = parse_args()?;
+    let service = Service::new_ref();
+    let js_ctx = service.context();
+    let js_args = args
+        .js_args
+        .to_js_value(&js_ctx)
+        .context("Failed to convert args to js value")?;
+    js_ctx
+        .get_global_object()
+        .set_property("scriptArgs", &js_args)
+        .context("Failed to set scriptArgs")?;
+    let mut expr_val = None;
+    for code in args.codes.into_iter() {
+        let result = match code {
+            JsCode::Source(src) => service.exec_script(&src),
+            JsCode::Bytecode(bytes) => service.exec_bytecode(&bytes),
+        };
+        match result {
+            Ok(value) => expr_val = value.to_js_value(),
+            Err(err) => {
+                bail!("Failed to execute script: {err}");
+            }
+        }
+    }
+    if service.number_of_tasks() > 0 {
+        service.wait_for_tasks().await;
+    }
+    // If scriptOutput is set, use it as output. Otherwise, use the last expression value.
```

Comment:
Just realized in an async program, the last expression value may not be the final return value of the program. Let's document it somewhere so people can get it.

Also, maybe call it `asyncScriptOutput` to address it?

Reply:
Personally I'd prefer to have a unified way to pass the output to avoid confusion to the user. I can imagine another choice is to always use the last expression result. It can pass a `Promise<T>` for async code. However, this would make the implementation much more complex.

Reply:
Yes, we should document it. I mean the rename is not necessary.

Reply:
Docmented it in the API doc of `js_eval`: https://github.com/Phala-Network/phala-blockchain/pull/1450/commits/3c1a3730f06fe68d4a1e165316ddb7fe6ae26e81